### ISSUE TITLE
SQS extra message received via SNS

### ISSUE THREAD

**Issue by rabberbock** [localstack/localstack#1516]
Running the following commands in order results in an extra message being put in the queue. Even though the topic is created twice this should not result in multiple messages being received by the subscribed queue.

I am using the latest release of localstack on docker. 
```
aws --region us-east-1  --endpoint-url http://localhost:4576 sqs create-queue --queue-name  raw-test
aws --region us-east-1  --endpoint-url http://localhost:4575 sns create-topic --name raw-test-topic
aws --region us-east-1  --endpoint-url http://localhost:4575 sns subscribe --topic-arn arn:aws:sns:us-east-1:000000000000:raw-test-topic --protocol sqs --notification-endpoint arn:aws:sqs:us-east-1:000000000000:raw-test --attributes RawMessageDelivery=true
aws --region us-east-1  --endpoint-url http://localhost:4575 sns publish --topic-arn arn:aws:sns:us-east-1:000000000000:raw-test-topic --message "hello"
aws --region us-east-1  --endpoint-url http://localhost:4576 sqs receive-message --queue-url http://localhost:4576/queue/raw-test
aws --region us-east-1  --endpoint-url http://localhost:4576 sqs receive-message --queue-url http://localhost:4576/queue/raw-test
aws --region us-east-1  --endpoint-url http://localhost:4575 sns create-topic --name raw-test-topic
aws --region us-east-1  --endpoint-url http://localhost:4575 sns publish --topic-arn arn:aws:sns:us-east-1:000000000000:raw-test-topic --message "hello"
aws --region us-east-1  --endpoint-url http://localhost:4576 sqs receive-message --queue-url http://localhost:4576/queue/raw-test
aws --region us-east-1  --endpoint-url http://localhost:4576 sqs receive-message --queue-url http://localhost:4576/queue/raw-test
```

Thanks for your help!

**Comment by menasheh** [comment]
Looks like this is a lower level, easier to understand demonstration of the same underlying problem as #933

**Comment by whummer** [comment]
Thanks for reporting @rabberbock . Looks like this should be fixed in #1649 - please re-open this issue if the problem persists. Thanks!
